src/Core/Checkout/Cart/Order/OrderConverter.php

```diff
@@ -173,6 +187,12 @@ public function convertToOrder(Cart $cart, SalesChannelContext $context, OrderCo
                 $this->initialStateIdLoader->get(OrderTransactionStates::STATE_MACHINE),
                 $context->getContext()
             );
+
+            if ($cart->getTransactions()->count() > 0) {
+                $primaryOrderTransactionId = Uuid::randomHex();
+                $data['transactions'][0]['id'] = $primaryOrderTransactionId;
+                $data['primaryOrderTransactionId'] = $primaryOrderTransactionId;
```

Comment:
we also need logic in the `SetPaymentOrderRoute`. The newly created transactions there (line 100 AFAIK) should become the new primaryTransaction, so After Order effectively works and the `last` stays the primary.